src/libs/antares/study/parameters.cpp

```diff
@@ -1020,10 +962,33 @@ bool Parameters::loadFromINI(const IniFile& ini, uint version, const StudyLoadOp
     reset();
     // A temporary buffer, used for the values in lowercase
     String value;
-
+    String sectionName;
+    typedef bool (*Callback)(Parameters&, const String&, const String&, const String&, const String&, uint);
+
+    static std::map<String,Callback> sectionAssociatedToKeysProcess = {
+            {"general",&SGDIntLoadFamily_General},
+            {"input",&SGDIntLoadFamily_Input},
+            {"output",&SGDIntLoadFamily_Output},
+            {"optimization",&SGDIntLoadFamily_Optimization},
+            {"other preferences",&SGDIntLoadFamily_OtherPreferences},
+            {"advanced parameters",&SGDIntLoadFamily_AdvancedParameters},
+            {"playlist",&SGDIntLoadFamily_Playlist},
+            {"variables selection",&SGDIntLoadFamily_VariablesSelection},
+            {"seeds",&SGDIntLoadFamily_Seeds}};
+
+    Callback* handleAllKeysInSection;
     // Foreach section on the ini file...
     for (auto* section = ini.firstSection; section; section = section->next)
     {
+        sectionName = section->name;
+        sectionName.toLower();
+        handleAllKeysInSection = &sectionAssociatedToKeysProcess[section->name]; // @TODO Must check that the function exists
```

Comment:
1. Use `std::map<T>::count` to determine the existence of the key within the map.
2. Use `sectionName` for the key.
3. Use `std::map<T>::at`, it won't try to add elements to your map unlike `operator[]`. More details on [at](https://www.cplusplus.com/reference/map/map/at/) and [operator\[\]](https://www.cplusplus.com/reference/map/map/operator[]/).

Reply:
Ok. Other solution adopted with try/catch block around .at use. If no section is found an exception, catcher, is raised. See the documentation if needed to confirm the try/catch block. 
See a8cb305